contracts/StakeVault.sol

```diff
@@ -9,64 +9,223 @@ import { StakeManager } from "./StakeManager.sol";
+        _withdraw(_token, _amount, msg.sender);
+    }
+
+    /**
+     * @notice Withdraw tokens from the contract to a destination address.
+     * @param _token The ERC20 token to withdraw.
+     * @param _amount The amount of tokens to withdraw.
+     * @param _destination The address to receive the tokens.
+     */
+    function withdraw(
+        ERC20 _token,
+        uint256 _amount,
+        address _destination
+    )
+        external
+        onlyOwner
+        validDestination(_destination)
+    {
+        _withdraw(_token, _amount, _destination);
+    }
+
+    /**
+     * @notice Withdraw Ether from the contract to the owner address.
+     * @param _amount The amount of Ether to withdraw.
+     */
+    function withdraw(uint256 _amount) external onlyOwner {
+        _withdraw(_amount, payable(msg.sender));
+    }
+
+    /**
+     * @notice Withdraw Ether from the contract to a destination address.
+     * @param _amount The amount of Ether to withdraw.
+     * @param _destination The address to receive the Ether.
+     */
+    function withdraw(
+        uint256 _amount,
+        address payable _destination
+    )
+        external
+        onlyOwner
+        validDestination(_destination)
+    {
+        _withdraw(_amount, _destination);
+    }
+
+    /**
+     * @notice Reject migration, leave staking contract and withdraw all tokens to the owner.
+     */
     function leave() external onlyOwner {
-        stakeManager.migrateTo(false);
-        STAKED_TOKEN.transferFrom(address(this), msg.sender, STAKED_TOKEN.balanceOf(address(this)));
+        _leave(msg.sender);
+    }
+
+    /**
+     * @notice Reject migration, leave staking contract and withdraw all tokens to a destination address.
+     * @param _destination The address to receive the tokens.
+     */
+    function leave(address _destination) external onlyOwner validDestination(_destination) {
+        _leave(_destination);
     }
 
     /**
      * @notice Opt-in migration to a new StakeManager contract.
+     * @dev Updates the stakeManager to the migrated contract.
      */
     function acceptMigration() external onlyOwner {
         StakeManager migrated = stakeManager.migrateTo(true);
         if (address(migrated) == address(0)) revert StakeVault__MigrationNotAvailable();
         stakeManager = migrated;
     }
 
+    /**
+     * @notice Returns the staked token.
+     * @return The ERC20 staked token.
+     */
     function stakedToken() external view returns (ERC20) {
         return STAKED_TOKEN;
     }
+
+    /**
+     * @notice Returns the available amount of a token that can be withdrawn.
+     * @param _token The ERC20 token to check.
+     * @return The amount of token available for withdrawal.
+     */
+    function availableWithdraw(ERC20 _token) external view returns (uint256) {
+        if (_token == STAKED_TOKEN) {
+            return STAKED_TOKEN.balanceOf(address(this)) - amountStaked;
+        }
+        return _token.balanceOf(address(this));
```

Comment:
In which case would we **not** want to get the amount from whatever `STAKED_TOKEN` is?

Reply:
Ah I see.. i think this is for when possibly, accidentially funds where sent to the stakevault?

Reply:
Yes, and can also be used as a feature to StakeVault receive rewards from StakeManager.